README.md

```diff
@@ -37,185 +41,198 @@ Simple parsing for node.js.
+Use `--help` with any of these commands to see available options.
 
-A parser consists of several *nonterminals*, which are constructions in a
-language. A nonterminal is made up of a series of either other nonterminals or
-strings. In nearley, you define a nonterminal by giving its name and its
-expansions.
+- `nearleyc` compiles grammar files to JavaScript.
+- `nearley-test` lets you quickly test a grammar against some input and see the
+  results. It also lets you explore the internal state of nearley's Earley
+  table, in case you find that interesting.
+- `nearley-unparse` inverts a parser into a generator, allowing you to create
+  random strings that match your grammar.
+- `nearley-railroad` generates pretty railroad diagrams from your parser. This
+  is mainly helpful for creating documentation, as (for example) on json.org.
+
+## Parser specification
+
+Let's explore the building blocks of a nearley parser.
+
+### Terminals, nonterminals, rules
+
+- A *terminal* is a string or a token. E.g. keyword `"if"` is a terminal.
+- A *nonterminal* is a combination of terminals and other nonterminals. E.g. an if statement defined as `"if" condition statement` is a nonteminal.
+- A *rule* (or production rule) is a definition of a nonterminal. E.g. `"if" condition statement` is the rule according to which the if statement nonterminal is parsed.
 
-Strings are the *terminals*, which match those string literals (specified as
-JSON-compatible strings).
+The first nonterminal of the grammar is the one the whole input must match. With the following grammar, nearley will try to parse text as `expression`.
 
-The following grammar matches a number, a plus sign, and another number:
+```js
+expression -> number "+" number
+number -> [0-9]:+
+```
 
-    expression -> number "+" number
-    
-Anything from a `#` to the end of a line is ignored as a comment:
+A nonterminal can have multiple alternative rules, separated by vertical bars (`|`):
 
-    expression -> number "+" number # sum of two numbers
+```js
+expression ->
+      number "+" number
+    | number "-" number
+    | number "*" number
+    | number "/" number
+```
 
-A nonterminal can have multiple expansions, separated by vertical bars (`|`):
+A special kind of rule is an **epsilon rule**. It matches nothing and is written as `null`. The following nonterminal matches zero or more `cow`s in a row:
 
-    expression ->
-          number "+" number
-        | number "-" number
-        | number "*" number
-        | number "/" number
+```js
+a -> null
+    | a "cow"
+```
 
-The parser tries to parse the first nonterminal that you define in a file.
-However, you can (and should!) introduce more nonterminals as "helpers". In
-this example, we would have to define the expansion of `number`.
+Keep in mind that nearley syntax is not sensitive to formatting. You may keep rules on the same line: `foo -> bar | qux`.
 
+### Comments
+
+Everything starting from `#` to the end of a line is ignored as a comment:
```

Comment:
"ignored as a comment" is slightly awkward. Perhaps just write "is ignored". :-)